test/org/sagebionetworks/bridge/dynamodb/DynamoScheduledActivityDaoTest.java

```diff
@@ -104,103 +102,92 @@ public void after() {
     @Test
     public void createUpdateDeleteActivities() throws Exception {
         DateTime endsOn = DateTime.now().plus(Period.parse("P4D"));
-        Map<String,DateTime> events = Maps.newHashMap();
-        events.put("enrollment", ENROLLMENT);
         
         ScheduleContext context = new ScheduleContext.Builder()
-            .withStudyIdentifier(TEST_STUDY)
+            .withUser(user)
             .withClientInfo(ClientInfo.UNKNOWN_CLIENT)
             .withTimeZone(DateTimeZone.UTC)
             .withEndsOn(endsOn)
-            .withHealthCode(user.getHealthCode())
-            .withEvents(events).build();
+            .withEvents(eventMap()).build();
         
-        List<ScheduledActivity> activitiesToSchedule = TestUtils.runSchedulerForActivities(user, context);
+        List<ScheduledActivity> activitiesToSchedule = TestUtils.runSchedulerForActivities(context);
         activityDao.saveActivities(activitiesToSchedule);
         
-        List<ScheduledActivity> activities = activityDao.getActivities(context);
-        int collectionSize = activities.size();
-        assertFalse("activities were created", activities.isEmpty());
+        List<ScheduledActivity> savedActivities = activityDao.getActivities(context.getZone(), activitiesToSchedule);
+        
+        assertEquals("activities were created", activitiesToSchedule, savedActivities);
         
-        // Should not increase the number of activities
-        activities = activityDao.getActivities(context);
-        assertEquals("activities did not grow afer repeated getActivity()", collectionSize, activities.size());
-
         // Have activities gotten injected time zone? We have to do this during construction using the time zone
         // sent with this call/request.
-        assertEquals(DateTimeZone.UTC, ((DynamoScheduledActivity)activities.get(0)).getTimeZone());
+        assertEquals(DateTimeZone.UTC, ((DynamoScheduledActivity)savedActivities.get(0)).getTimeZone());
```

Comment:
In a single request you can't do this and I'm not sure the results would be accurate or desirable. The ScheduledActivityServiceTest.retrievalActivitiesAcrossTimeAndTimeZones() test examines calling getActivities() from different time zones, e.g. if the user took a plane to a different time zone and opened the app, and the behavior is consistent.